remoulade/middleware/pipelines.py

```diff
@@ -37,20 +41,83 @@ def actor_options(self):
+
+        If it's a group, we need to get the result of each actor from the result backend because it's not present in
+        the message.
+
+        Parameters:
+            pipe_target(dict|List[dict]): the message or list of message to enqueue
+            broker(Broker): the broker
+            result(any): the result of the actor (to be send with the message if pipe_ignore=False)
+            group_info(GroupInfo|None): the info of the group if it's a group else None
+        """
         # Since Pipelines is a default middleware, this import has to
         # happen at runtime in order to avoid a cyclic dependency
         # from broker -> pipelines -> messages -> broker.
         from ..message import Message
 
-        if exception is not None:
+        if isinstance(pipe_target, list):
+            for message_data in pipe_target:
+                self._send_next_message(message_data, broker, result, group_info)
             return
 
-        message_data = message.options.get("pipe_target")
-        if message_data is not None:
-            next_message = Message(**message_data)
-            next_actor = broker.get_actor(next_message.actor_name)
-            pipe_ignore = next_message.options.get("pipe_ignore") or next_actor.options.get("pipe_ignore")
-            if not pipe_ignore:
-                next_message = next_message.copy(args=next_message.args + (result,))
+        next_message = Message(**pipe_target)
+        next_actor = broker.get_actor(next_message.actor_name)
+        pipe_ignore = next_message.options.get("pipe_ignore") or next_actor.options.get("pipe_ignore")
+
+        if not pipe_ignore:
+            if group_info:
+                result = self._get_group_result(group_info)
+            next_message = next_message.copy(args=next_message.args + (result,))
+
+        broker.enqueue(next_message)
+
+    @staticmethod
+    def _group_completed(group_info, broker):
+        """ Returns true if a group is completed, and increment the completion count of the group
+
+        Parameters:
+            group_info(GroupInfo): the info of the group to get the completion from
+            broker(Broker): the broker to use
+
+        Raises:
+            NoResultBackend: if there is no result backend set
+        """
+        try:
+            result_backend = broker.get_result_backend()
+        except NoResultBackend:
+            raise NoResultBackend('Pipeline with groups are ony supported with a result backend')
+
+        group_completion = result_backend.increment_group_completion(group_info.group_id)
+
+        return group_completion >= group_info.count
+
+    @staticmethod
+    def _get_group_result(group_info):
+        if group_info.results is None:
+            raise ResultNotStored('An actor in the group do not have store_result=True')
 
-            broker.enqueue(next_message)
+        return list(group_info.results.get(forget=True))
```

Comment:
Should we forget here ? this means there is no way to check progress anymore. I should do a progress object, but we are a little short here, I'll do it later